subsys/bluetooth/audio/ascs.c

```diff
@@ -2376,10 +2376,14 @@ static void ase_stop(struct bt_ascs_ase *ase)
 	 * for that ASE by following the Connected Isochronous Stream Terminate
 	 * procedure defined in Volume 3, Part C, Section 9.3.15.
 	 */
-	err = ascs_disconnect_stream(stream);
-	if (err < 0) {
-		LOG_ERR("Failed to disconnect stream %p: %d", stream, err);
-		return;
+	if (ep->iso != NULL &&
+	    ep->iso->chan.state != BT_ISO_STATE_DISCONNECTED &&
+	    ep->iso->chan.state != BT_ISO_STATE_DISCONNECTING) {
+		err = ascs_disconnect_stream(stream);
+		if (err < 0) {
+			LOG_ERR("Failed to disconnect stream %p: %d", stream, err);
+			return;
+		}
```

Comment:
For the unicast client we actually have a check like `unicast_client_can_disconnect_stream` which is similar to this, but also takes into account that a CIS may be used by 2 audio streams. I guess that last part might still be missing here, right? i.e. if the CIS is used for another stream in the other direction that is still streaming, we should not attempt to disconnect the CIS here, right?